test/e2e/common/node/security_context.go

```diff
@@ -633,6 +639,249 @@ var _ = SIGDescribe("Security Context", func() {
+						}
+					})
+					ginkgo.It("it should add SupplementalGroups to them [LinuxOnly]", func(ctx context.Context) {
+						expectMergePolicyInEffect(ctx, f, pod.Name, pod.Spec.Containers[0].Name, true)
+					})
+				})
+			})
+		})
+		ginkgo.When("SupplementalGroupsPolicy was set to Merge in PodSpec", func() {
+			ginkgo.When("the container's primary UID belongs to some groups in the image", func() {
+				var pod *v1.Pod
+				ginkgo.BeforeEach(func(ctx context.Context) {
+					ginkgo.By("creating a pod", func() {
+						pod = e2epod.NewPodClient(f).CreateSync(ctx, mkPod(nil))
+					})
+				})
+				ginkgo.When("scheduled node does not support SupplementalGroupsPolicy", func() {
+					ginkgo.BeforeEach(func(ctx context.Context) {
+						// ensure the scheduled node does not support SupplementalGroupsPolicy
+						if nodeSupportsSupplementalGroupsPolicy(ctx, f, pod.Spec.NodeName) {
+							e2eskipper.Skipf("scheduled node does support SupplementalGroupsPolicy")
+						}
+					})
+					ginkgo.It("it should add SupplementalGroups to them [LinuxOnly]", func(ctx context.Context) {
+						expectMergePolicyInEffect(ctx, f, pod.Name, pod.Spec.Containers[0].Name, false)
+					})
+				})
+				ginkgo.When("scheduled node supports SupplementalGroupsPolicy", func() {
+					ginkgo.BeforeEach(func(ctx context.Context) {
+						// ensure the scheduled node does support SupplementalGroupsPolicy
+						if !nodeSupportsSupplementalGroupsPolicy(ctx, f, pod.Spec.NodeName) {
+							e2eskipper.Skipf("scheduled node does not support SupplementalGroupsPolicy")
+						}
+					})
+					ginkgo.It("it should add SupplementalGroups to them [LinuxOnly]", func(ctx context.Context) {
+						expectMergePolicyInEffect(ctx, f, pod.Name, pod.Spec.Containers[0].Name, true)
+					})
+				})
+			})
+		})
+		ginkgo.When("SupplementalGroupsPolicy was set to Strict in PodSpec", func() {
+			ginkgo.When("the container's primary UID belongs to some groups in the image", func() {
+				var pod *v1.Pod
+				ginkgo.BeforeEach(func(ctx context.Context) {
+					ginkgo.By("creating a pod", func() {
+						pod = e2epod.NewPodClient(f).Create(ctx, mkPod(ptr.To(v1.SupplementalGroupsPolicyStrict)))
+						framework.ExpectNoError(e2epod.WaitForPodScheduled(ctx, f.ClientSet, pod.Namespace, pod.Name))
+						var err error
+						pod, err = e2epod.NewPodClient(f).Get(ctx, pod.Name, metav1.GetOptions{})
+						framework.ExpectNoError(err)
+					})
+				})
+				ginkgo.When("scheduled node does not support SupplementalGroupsPolicy", func() {
+					ginkgo.BeforeEach(func(ctx context.Context) {
+						// ensure the scheduled node does not support SupplementalGroupsPolicy
+						if nodeSupportsSupplementalGroupsPolicy(ctx, f, pod.Spec.NodeName) {
+							e2eskipper.Skipf("scheduled node does support SupplementalGroupsPolicy")
+						}
+					})
+					ginkgo.It("it should reject the pod [LinuxOnly]", func(ctx context.Context) {
+						expectRejectionEventIssued(ctx, f, pod)
+					})
+				})
```

Comment:
This e2e case requires old CRI runtimes (contaienrd v1.x or CRI <1.31). But it seems that we don't have such CI jobs currently. 

So, I have attached my local E2E test result (with containerd v1.7) in the [PR description(see "Special notes for your reviewer" section)](https://github.com/kubernetes/kubernetes/pull/130210).

If you think we need to define a dedicated CI job to test this feature on containerd v1.x (which does not support this feature). I would be glad if you gave me some advice.